api/src/devcomp/domain/services/solution-service-qrocm-ind.js

```diff
@@ -36,8 +35,15 @@ function match({ answerValue, solution }) {
   };
 }
 
+function _isEmpty(object) {
+  return !object || Object.keys(object).length === 0;
+}
+function _isString(value) {
+  return !!value && (typeof value === 'string' || value instanceof String);
+}
+
 function _applyTreatmentsToSolutions(solutions, enabledTreatments, qrocBlocksTypes = {}) {
-  return _.forEach(solutions, (solution, solutionKey) => {
+  Object.entries(solutions).forEach(([solutionKey, solution]) => {
```

Comment:
OK pour moi de virer Lodash.

Par contre, je n'ai pas l'impression que les `Object.entries` `forEach` améliorent la lisibilité. À titre perso, je préfère en général des [fonctions pures](https://fr.wikipedia.org/wiki/Fonction_pure).

Ici par exemple j'ai le sentiment qu'on veut faire du `map` mais la fonction reste très difficile à lire ligne par ligne.

Sans trop creuser, est-ce que des simples `for` ne seraient pas plus simple dans certaines méthodes de ce fichier ? Est-ce qu'on ne veut pas extraire certaines fonctions pour les tester unitairement ?

Reply:
En vrai je suis assez d'accord : on est en train de reconstruire un nouvel objet Solutions, donc soit un map, soit finalement un reduce. Je vais tester le truc

Reply:
> Est-ce qu'on ne veut pas extraire certaines fonctions pour les tester unitairement ?

Les fonctions en question sont déjà testées unitairement ici `api/tests/devcomp/unit/domain/services/solution-service-qrocm-ind_test.js` ;) mais à voir si c'est suffisant.

Penses-tu que mes fonctions `isEmpty` et `isString` doivent être testées unitairement ? Ou bien comme elles sont localisées dans le solution-service, on peut considérer que leur validation est portée par l'ensemble des autres tests ?

Reply:
c'est effectivement assez confus:

```javascript
function _applyTreatmentsToSolutions(solutions, enabledTreatments, qrocBlocksTypes = {}) {
  return Object.entries(solutions).reduce((s, [solutionKey, solution]) => {
    s[solutionKey] = solution;
    solution.forEach((variant, variantIndex) => {
      if (qrocBlocksTypes[solutionKey] === 'select') {
        s[solutionKey][variantIndex] = applyTreatments(variant, []);
      } else {
        s[solutionKey][variantIndex] = applyTreatments(variant, enabledTreatments);
      }
    });
    return s;
  }, {});
}
```

Je tente le map

Reply:
Le soucis du map c'est qu'il retourne un tableau alors que nous, nous retournons un objet à la fin